_posts/jared_baker/2018-12-15-Bluestore.md

```diff
@@ -0,0 +1,171 @@
+---
+layout: post
+title:  "Migrating 8 PB of data from Filestore to Bluestore"
+breadcrumb: true
+author: jared_baker
+date: 2018-12-15
+categories: jared_baker
+tags:
+    - ceph
+    - linux
+    - bluestore
+    - tutorial
+teaser:
+    info: How we migrated an 8 PB cluster from Filestore to Bluestore
+    image: jared_baker/bluestore/Ceph_Logo_Standard_RGB_120411_fa.png
+header:
+    version: small
+    title: Software Engineering Blog
+    image: jared_baker/le-logo-standard.png
+    icon: icon-blog
+---
+## Introduction
+The Cancer Genome Collaboratory is a Cloud computing environment designed for cancer research and an official repository of 950 TB (and growing) worth of whole genomes for the [ICGC project](https://dcc.icgc.org/repositories). We have used OpenStack and Ceph to provide an extremely scalable and reliable infrastructure. Maintaining the pieces namely through upgrades is our most important and time consuming task in operating a cloud like this. Whether you are chasing new features, bug fixes, security patches or supportability it seems like we are always planning for the next upgrade.
+
+## About our Ceph environment
+Our Ceph cluster was born in 2014 and was built upon Ceph Giant and eventually upgraded to Hammer, then to Ceph Jewel and more recently this year to Luminous. Up until Luminous our OSD's were using the only available osd backend at the time, Filestore, with xfs and co-located journals and 3 replicas for data redundancy. Our number of OSD's has been growing steadily throughout the years to maximize purchase value per terabyte and spreading our budget over a number of years so that hardware purchases only occurred when capacity was forecasted.
+
+We currently have 37 storage nodes, each with 36 drives for a total of 1332 OSD's. As mentioned before, purchasing nodes over time means we have a variety of HDD sizes ranging from 4 TB to 12 TB. Our smallest nodes are 144 TB while our largest are 432 TB. Each storage node has 2 separate SSD's for the OS, 256 GB of RAM, dual Xeon CPU's and 40 Gbps of network capability through the use of vlan bonds to partition the traffic type (ceph public, replication, etc).
+
+The primary use case for our Ceph Cluster is the storage of those very large genomic files and are served to researchers using the rados gateway object storage api. We do also supplement the OpenStack environment with volume support using cinder but it amounts to less than 1% of our total utilization of the ceph cluster.
+
+## Bluestore in Ceph Luminous
+[Bluestore](https://ceph.com/community/new-luminous-bluestore/) is the new default storage backend starting in Ceph Luminous and proposes to improve write performance by a factor of 2, and sometimes higher in certain scenarios by removing the double write penalty that took place on Filestores journal and xfs partition. Most notably Bluestore removes the requirement of a posix filesystem to store ceph data. See the diagram below for the architectural differences.
```

Comment:
- Add apostrophe to '_Filestore's_'
- Should 'xfs' and 'posix' be all-caps?
- Capitalize last occurrence of 'ceph' to be consistent with rest of post